src/components/TodoList/TodoList.tsx

```diff
@@ -1,210 +1,111 @@
 /* eslint-disable max-len */
-import React from 'react';
+import {
+  FC, useMemo,
+} from 'react';
+import cn from 'classnames';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { Todo } from '../../types/Todo';
+import { actions as currentTodoActions } from '../../features/currentTodo';
+import { filterTodos } from '../../helpers/filterFunctionForTodos';
+
+export const TodoList: FC = () => {
+  const dispatch = useAppDispatch();
+  const todos = useAppSelector(state => state.todos);
+  const selectedTodo = useAppSelector(state => state.currentTodo);
+
+  const { query, status } = useAppSelector(state => state.filter);
+
+  const handleTodoSelection = (todo: Todo) => {
+    dispatch(currentTodoActions.setTodo(todo));
+  };
+
+  const visibleTodos = useMemo(
+    () => filterTodos(todos, query, status),
+    [todos, query, status],
+  );
```

Comment:
It could be a custom hook that doesn't take arguments and returns visibleTodos
```
const useVisibleTodos = () => {
  // call selectors

 // return useMemo resuls
}